.github/ariane-config.yaml

```diff
@@ -4,22 +4,18 @@ allowed-teams:
 triggers:
   /test:
     workflows:
-    - conformance-aws-cni.yaml
     - conformance-clustermesh.yaml
     - conformance-delegated-ipam.yaml
     - conformance-ipsec-e2e.yaml
-    - conformance-eks.yaml
     - conformance-externalworkloads.yaml
     - conformance-ginkgo.yaml
     - conformance-ingress.yaml
     - conformance-multi-pool.yaml
     - conformance-runtime.yaml
     - integration-test.yaml
-    - tests-clustermesh-upgrade.yaml
```

Comment:
Please note that the clustermesh upgrade tests are currently run on every push to main. If the main concern here is the cost and/or amount of resources that is consumed by these tests, then that's something to reason about as well. Otherwise, this specific test has proven to be reasonably stable, and typically requires less than 30 minutes (although the duration seems to have increased over time due to the Cilium agents taking longer during rollout).

Reply:
I think there's some balance to be found between the cost and the valuable signal out of a particular testsuite. For many of the changes we make in Cilium, they probably aren't going to impact clustermesh upgrade. This can mean we get effectively zero signal across a large number of runs. However, probably when making changes to some specific paths in the codebase, they have a higher probability of regression. For those paths, we should run these tests before merging, and we can do this manually via a dedicated test command, or we can try to conditionally run the workflow. Where that line is exactly is up for discussion.

If we think there's a reasonable regex we can add towards the end of the file to narrow down changes that are likely to impact this test, then that'd be nice to include. Otherwise I think it's worth thinking about whether to run these primarily upon merge and/or schedule and then monitor them for failures post-merge.

Reply:
> For many of the changes we make in Cilium, they probably aren't going to impact clustermesh upgrade. This can mean we get effectively zero signal across a large number of runs.

Agreed. Although I'd argue that the same also applies to the runs triggered post merge, which compared to these ones don't give any direct signal to developers about whether their changes are breaking a workflow or not. I don't know how many times on average any given workflow gets run for each PR to compare the impact of the two though.

> However, probably when making changes to some specific paths in the codebase, they have a higher probability of regression. For those paths, we should run these tests before merging, and we can do this manually via a dedicated test command, or we can try to conditionally run the workflow.

Besides the obvious ones (any clustermesh change in this case), in my experience it is quite difficult to foresee in advance whether a change will have an impact on a given workflow or not, especially the ones which do test upgrades and downgrades and assert that connections are not disrupted. Leave it alone that many contributors may not be even aware of clustermesh or its implications. Realistically, I do think that non-required workflows will be manually run on a very minority of the PRs. 

> If we think there's a reasonable regex we can add towards the end of the file to narrow down changes that are likely to impact this test, then that'd be nice to include.

At the very minimum, it would be good to have this workflow run on any change touching files in the clustermesh packages. However, my understanding is that this is not possible with Ariane as of today. While we can specify a regex to narrow down the changes that trigger a given workflow, that's independent of the trigger phrase, making then impossible to manually run it if none of the files are matched.

While I'm not strongly opposed to this change, my main concern relates to the lack of clarity on how post merge failures will be handled, and how much additional load will that add (in this case) to the clustermesh team to triage and potentially address them. If any possible failure of this workflow needs to escalated to the team, then this is not going to scale well, and the workflow will just get in a worse and worse shape over time -- it is not a PR blocker after all, and therefore gets lower overall attention compared to the other flakes.

Reply:
on.push events are tracked as part of the monitoring of CI flakes. If a certain workflow/job/step fails often, it will be flagged and raised for fixing.

To Joe's suggestion, there is a `paths` option for on.push events - if we know which folders CI should monitor for changes, we can trigger workflows runs when changes have been committed on certain folders (e.g. `.github/workflows/tests-clustermesh-upgrade.yaml` and `clustermersh/`).

https://docs.github.com/en/actions/writing-workflows/workflow-syntax-for-github-actions#onpushpull_requestpull_request_targetpathspaths-ignore

Reply:
> on.push events are tracked as part of the monitoring of CI flakes. If a certain workflow/job/step fails often, it will be flagged and raised for fixing.

Sure. What's unclear to me is who will be responsible for fixing the issues, as IMO that's not going to scale if all the load will be on the clustermesh team (in this specific example, but the same applies to say the ipsec team for the ipsec workflow). It is already common today that optional workflows are persistently in a worse situation compared to required ones (and indeed they have been removed altogether), as there's very little incentive in triaging and fixing the associated flakes, and the overall effort is typically prioritized towards blocking ones (for obvious reasons).

> To Joe's suggestion, there is a paths option for on.push events - if we know which folders CI should monitor for changes, we can trigger workflows runs when changes have been committed on certain folders (e.g. .github/workflows/tests-clustermesh-upgrade.yaml and clustermersh/).

IMO it would be more valuable if the tests were run directly upon `/test` when files in relevant folders (e.g., `pkg/clustermesh`) are modified, to try to catch the problems before actually merging the given PR. That seems to me an extension of the current path filter support in Ariane, so that, when running `/test`, a given optional workflow is triggered only if the specified subset of files is modified, while it is always run (excluding e.g., documentation only changes) if explicitly triggered via say `/ci-clustermesh`. WDYT?

Reply:
> What's unclear to me is who will be responsible for fixing the issues

I agree that driving the responsibility aspect to the next level of detail would be important to be able to convince ourselves that we're not going to cause a lot of pain for developers with this kind of change. If Isovalent is planning to nominate some people to monitor this, notify developers, revert broken PRs then great - though I would encourage building an open process that allows non-Isovalent contributors to participate too. We can also just start with _some_ process and try it out, then iterate from there. @auriaave feel free to reach out to me if you'd like to brainstorm together on that.

> IMO it would be more valuable if the tests were run directly upon /test when files in relevant folders (e.g., pkg/clustermesh) are modified, to try to catch the problems before actually merging the given PR.

I think this is how the Ariane configuration works today - if there's a file match pattern for the workflow, then running `/test` will trigger the test to run. If the file match pattern doesn't match, Ariane will skip the workflow. The part I'm not sure about is.. what if it's skipped by default, but we actually want to bypass the file matcher and force-run the test, because for instance we're not sure if the PR may break clustermesh. Can we just run `/ci-clustermesh` on any PR and know that the CI will force-run that test workflow?

Reply:
> What's unclear to me is who will be responsible for fixing the issues

I agree that driving the responsibility aspect to the next level of detail would be important to be able to convince ourselves that we're not going to cause a lot of pain for developers with this kind of change. If Isovalent is planning to nominate some people to monitor this, notify developers, revert broken PRs then great - though I would encourage building an open process that allows non-Isovalent contributors to participate too. We can also just start with _some_ process and try it out, then iterate from there. @auriaave feel free to reach out to me if you'd like to brainstorm together on that.

> IMO it would be more valuable if the tests were run directly upon /test when files in relevant folders (e.g., pkg/clustermesh) are modified, to try to catch the problems before actually merging the given PR.

I think this is how the Ariane configuration works today - if there's a file match pattern for the workflow, then running `/test` will trigger the test to run. If the file match pattern doesn't match, Ariane will skip the workflow. The part I'm not sure about is.. what if it's skipped by default, but we actually want to bypass the file matcher and force-run the test, because for instance we're not sure if the PR may break clustermesh. Can we just run `/ci-clustermesh` on any PR and know that the CI will force-run that test workflow?

Reply:
> The part I'm not sure about is.. what if it's skipped by default, but we actually want to bypass the file matcher and force-run the test, because for instance we're not sure if the PR may break clustermesh. Can we just run /ci-clustermesh on any PR and know that the CI will force-run that test workflow?

Yeah, this is the part I don't think is possible today, as the path regex is associated with the workflow, not a specific trigger phrase: https://github.com/cilium/cilium/blob/4a995ead80044ae4ab5c254f1b2de2b96f20a52b/.github/ariane-config.yaml#L127-L128

Reply:
> Yeah, this is the part I don't think is possible today, as the path regex is associated with the workflow, not a specific trigger phrase:

As @giorio94 said force running a workflow is not possible. There is an option to provide arguments to issue comments like `/ci-clustermesh force` however these arguments are only passed to the workflow. We could, in the same spirit, add `force` as a special argument that runs the workflow regardless of the changed paths.

Reply:
@giorio94 @joestringer do we have a conclusion? It is not clear to me how to go forward.

Reply:
IMO, the main open point is still around who will be responsible for triaging and fixing the issues that will inevitably start popping up due to these tests not being run on pre-merge for every PR, unless I missed the conclusion there.

Reply:
Triager here, any progress on this?

Reply:
Sorry, collecting information about the workflow run history took a little time. The data collected is for the last 3 months. About %8.05 of tests failed, %3.69 push, and %4.36  workflow_dispatch events failed. Unfortunately, we do not have the data about how many of these failures are flakes. IMO success rate for this workflow is good enough that failures in scheduled events can be triaged by CI-Health rotation. If something creeps in that brakes the tests then it can reverted just like any failure introduced, the author needs to fix it and test it manually before merging.

The average runtime per day is 15 hours, 27 minutes

Run result
| conclusion | count(conclusion) |
| --------- | --------- |
| failure | 218 |
| cancelled | 74 |
| success | 2708 |

Failed run distribution by workflow_dispatch event.
| head_branch | total |
| --------- | --------- |
| main | 57 |
| v1.15 | 14 |
| v1.16 | 13 |
| v1.14 | 11 |
| pr/tammach/test-dummy | 2 |
| pr/learnitall/ginkgo-race-workflow | 2 |
| pr/tammach/k8s-upgrade | 2 |
| pr/tammach/ingress-port-80 | 1 |
| pr/tklauser/prepare-for-golangci-lint-1.60.1 | 1 |
| pr/tammach/internal-address-config-1.14 | 1 |
| pr/tp/ci-runtime-allow-extra-cilium-args | 1 |
| pr/vk/eks/tests/concurrent/run | 1 |
| pr/remove-1.13-references | 1 |
| pr/tklauser/images-update-dont-assume-filename | 1 |
| pr/prepare-v1.16.1 | 1 |
| pr/v1.16-backport-2024-09-13-01-43 | 1 |
| pr/v1.16-backport-2024-10-25-02-11 | 1 |
| pr/bump-time-out-runtime | 1 |
| pr/tklauser/update-go-1.23 | 1 |
| pr/michi/new-branch | 1 |
| pr/tklauser/stdlib-slices | 1 |
| pr/tammach/cleanup-disk-build | 1 |

Failed run distribution by push event.
| head_branch | total |
| --------- | --------- |
| main | 22 |
| renovate/main-all-dependencies | 20 |
| renovate/v1.16-go | 9 |
| renovate/main-go | 9 |
| renovate/v1.16-all-dependencies | 8 |
| v1.15 | 7 |
| v1.16 | 6 |
| renovate/v1.15-patch-stable-lvh-images | 5 |
| renovate/v1.16-patch-stable-lvh-images | 4 |
| renovate/v1.14-patch-stable-lvh-images | 3 |
| renovate/main-all-github-action | 2 |
| renovate/v1.14-all-github-action | 2 |
| renovate/main-aws-sdk-go-v2-monorepo | 1 |
| renovate/v1.16-all-github-action | 1 |
| renovate/main-all-go-deps-main | 1 |
| renovate/main-patch-all-lvh-images-main | 1 |
| renovate/v1.14-all-dependencies | 1 |